### PR TITLE
try minimal requirements

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

unfreeze requirements, cc: @SeanNaren 

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning-lash/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? [not needed for typos/docs]
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by carmocca** [comment]
I'm not sure we can have `torch<1.7.1` because for earlier versions, the loaded weights in the examples will fail due to PyTorch breaking compatibility in their serialization

**Comment by Borda** [comment]
> I'm not sure we can have `torch<1.7.1` because for earlier versions, the loaded weights in the examples will fail due to PyTorch breaking compatibility in their serialization

good point but in such case, we shall have it tested somewhere... anyway fix free on one particular version is very bad :/

**Comment by carmocca** [comment]
We have a test for it

https://github.com/PyTorchLightning/lightning-flash/blob/3c5b535ce26a12fd141acff7e20bc2b18ee56a0a/tests/core/test_model.py#L115-L129

We would need to run it with different PyTorch versions

**Comment by Borda** [comment]
> I'm not sure we can have `torch<1.7.1` because for earlier versions, the loaded weights in the examples will fail due to PyTorch breaking compatibility in their serialization

but this breaking change was in 1.6? @SeanNaren

**Comment by carmocca** [comment]
Yes, the change was in 1.6 but we had some issues before release and fixed it to 1.7.1 to be sure.

Can you add a single job with PyTorch 1.6 in CI? To make sure

**Comment by Borda** [comment]
> Can you add a single job with PyTorch 1.6 in CI? To be sure

now the min is 1.6 so all 6 tests are using it - Linux/Mac/Win and py3.6 and py3.8

**Review by tchaton — APPROVED** [review]

**Review by tchaton — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Review by carmocca — DISMISSED** [review]
We need to think about how to solve the weights compatibility issue. See failing tests:

https://github.com/PyTorchLightning/lightning-flash/blob/90a5d798d7d494cc66ca3e9b1294679769095f32/tests/core/test_model.py#L115-L129

`RuntimeError: Only one file(not dir) is allowed in the zipfile`


The weights were generated with `torch==1.7.1` and fail to load with `torch==1.6.0`

**Comment by Borda** [comment]
> The weights were generated with `torch==1.7.1` and fail to load with `torch==1.6.0`

Yes, the weights shall be generated in 1.6 so they would be compatible with all after 1.6...

**Comment by carmocca** [comment]
> Yes, the weights shall be generated in 1.6 so they would be compatible with all after 1.6...

That's what we thought initially but didn't seem to work before release. We'll have to try again

**Comment by Borda** [comment]
@carmocca it is also what I did for PL checkpoint legacy testing, all are created with PT 1.4 as they are possible to load in any future version up to 1.8

**Comment by Borda** [comment]
@carmocca how is it going with generating weights with lower pytorch?

File: requirements.txt [hunk 1]
```diff
@@ -1,14 +1,13 @@
 pytorch-lightning==1.2.0rc0
-torch==1.7.1
-PyYAML==5.3.1
+torch>=1.7  # regenerate weigths with lewer PT version
```
Comment (Borda) [inline requirements.txt:2]:
```suggestion
torch>=1.7  # TODO: regenerate weigths with lewer PT version
```

File: requirements.txt [hunk 2]
```diff
@@ -1,14 +1,13 @@
...
+PyYAML>=5.1
 Pillow>=7.2
-torchvision==0.8.2
-transformers==4.2.2
-pytorch-tabnet==3.1.1
-datasets==1.2.1
-pandas==1.1.5 ; python_version < "3.7"
-pandas==1.2.2 ; python_version >= "3.7"
-scikit-learn==0.24.0
+torchvision>=0.8.2  # lower to 0.7 after PT 1.6
```
Comment (Borda) [inline requirements.txt:5]:
```suggestion
torchvision>=0.8  # lower to 0.7 after PT 1.6
```

**Comment by carmocca** [comment]
> @carmocca how is it going with generating weights with lower pytorch?

Bit busy with the 1.2 release. @teddykoker can you do it? I believe you did it originally

**Review by carmocca — APPROVED** [review]

File: requirements.txt [hunk 3]
```diff
@@ -1,14 +1,13 @@
...
-pytorch-tabnet==3.1.1
-datasets==1.2.1
-pandas==1.1.5 ; python_version < "3.7"
-pandas==1.2.2 ; python_version >= "3.7"
-scikit-learn==0.24.0
+torchvision>=0.8  # lower to 0.7 after PT 1.6
+transformers>=4.0
+pytorch-tabnet>=3.0
+datasets>=1.2
+pandas>=1.0
```
Comment (Borda) [inline requirements.txt:9]:
```suggestion
pandas>=1.1
```

File: requirements.txt [hunk 4]
```diff
@@ -1,14 +1,13 @@
...
-torchvision==0.8.2
-transformers==4.2.2
-pytorch-tabnet==3.1.1
-datasets==1.2.1
-pandas==1.1.5 ; python_version < "3.7"
-pandas==1.2.2 ; python_version >= "3.7"
-scikit-learn==0.24.0
+torchvision>=0.8  # lower to 0.7 after PT 1.6
+transformers>=4.0
+pytorch-tabnet>=3.0
```
Comment (Borda) [inline requirements.txt:7]:
```suggestion
pytorch-tabnet==3.1.1
```

**Comment by Borda** [comment]
> > @carmocca how is it going with generating weights with lower pytorch?
> 
> Bit busy with the 1.2 release. @teddykoker can you do it? I believe you did it originally

no rush, I have created an issue for it... #128

File: requirements.txt [hunk 5]
```diff
@@ -1,14 +1,13 @@
...
-torchvision==0.8.2
-transformers==4.2.2
+torchvision>=0.8  # lower to 0.7 after PT 1.6
+transformers>=4.0
 pytorch-tabnet==3.1.1
-datasets==1.2.1
-pandas==1.1.5 ; python_version < "3.7"
-pandas==1.2.2 ; python_version >= "3.7"
-scikit-learn==0.24.0
+datasets>=1.2
```
Comment (Borda) [inline requirements.txt:8]:
```suggestion
datasets>=1.2, <1.3
```

File: requirements.txt [hunk 6]
```diff
@@ -1,14 +1,13 @@
...
-transformers==4.2.2
-pytorch-tabnet==3.1.1
-datasets==1.2.1
-pandas==1.1.5 ; python_version < "3.7"
-pandas==1.2.2 ; python_version >= "3.7"
-scikit-learn==0.24.0
+torchvision>=0.8  # lower to 0.7 after PT 1.6
+transformers>=4.0
+pytorch-tabnet==3.1
+datasets>=1.2, <1.3
```
Comment (Borda) [inline requirements.txt:8]:
```suggestion
datasets==1.2
```

**Comment by Borda** [comment]
ok, let's try reverse strategy :D

**Comment by Borda** [comment]
The Win-latest is failing also on master - https://github.com/PyTorchLightning/lightning-flash/runs/1936420296

**Pull request merged**